fhir2dataset/parser.py

```diff
@@ -0,0 +1,193 @@
+            self.config["join"]["parent"] = dict(self.__parent_join)
+        # print(f"self.config: {self.config}")
+
+        keys_to_pop = []
+        for key, value in self.config.items():
+            if not value:
+                keys_to_pop.append(key)
+        # print(f"keys_to_pop: {keys_to_pop}")
+        for key in keys_to_pop:
+            self.config.pop(key)
+
+    def __select_parser(self, string):
+        item_parsed = re.split(self.__split_mask_select, string)
+        # print("\nSELECT")
+        # print(f"item_parsed: {item_parsed}")
+        for item in item_parsed:
+            alias, select_rule = re.split(r"\.", item, 1)
+            assert alias in self.__from.keys()
+            self.__select[alias].append(select_rule)
+        # print(f"self.__select: {self.__select}")
+
+    def __from_parser(self, string):
+        item_parsed = re.split(self.__split_mask_from, string)
+        # print("\nFROM")
+        # print(f"item_parsed: {item_parsed}")
+        if len(item_parsed) == 2:
+            resource_type = item_parsed[0]
+            alias = item_parsed[1]
+            self.__from[alias] = resource_type
+        elif len(item_parsed) == 1:
+            resource_type = item_parsed[0]
+            alias = item_parsed[0]
+            self.__from[alias] = resource_type
+        else:
+            raise ValueError
+        # print(f"self.__from: {self.__from}")
+
+    def __inner_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__inner_join[alias_parent][searchparam_parent] = alias_child
+
+    def __child_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__child_join[alias_parent][searchparam_parent] = alias_child
+
+    def __parent_join_parser(self, string):
+        alias_parent, searchparam_parent, alias_child = self.__join_parser(string)
+        self.__parent_join[alias_parent][searchparam_parent] = alias_child
+
+    def __join_parser(self, string):
+        item_parsed = re.split(self.__split_mask_from_join, string)
+        # print("\nJOIN")
+        # print(f"item_parsed: {item_parsed}")
+        assert len(item_parsed) == 2
+        self.__from_parser(item_parsed[0])
+
+        join_conditions = re.split(self.__split_mask_join, item_parsed[1])
+        assert len(join_conditions) == 2
+
+        alias_parent = None
+        searchparam_parent = None
+        alias_child = None
+        for join_condition in join_conditions:
+            alias, join_rule = re.split(r"\.", join_condition, 1)
+            assert alias in self.__from.keys(), (
+                f"The {alias} alias used in the {join_condition} join must have been referenced "
+                "behind a FROM, INNER JOIN, PARENT JOIN or CHILD JOIN. "
+                f"The only aliases referenced are here: {self.__from}"
+            )
+            if join_rule in ["id", "id ", " id", " id "]:
```

Comment:
can you use `if join_rule.trim() == "id"` instead ?

Reply:
Good catch! It's changed :+1: